src/components/chats/ChatItem/Embed.tsx

```diff
@@ -0,0 +1,121 @@
+import LinkText from '@/components/LinkText'
+import MediaLoader from '@/components/MediaLoader'
+import { cx } from '@/utils/class-names'
+import { LinkMetadata } from '@subsocial/api/types'
+import truncate from 'lodash/truncate'
+import { ComponentProps, useMemo } from 'react'
+import {
+  InstagramEmbed,
+  TikTokEmbed,
+  YouTubeEmbed,
+} from 'react-social-media-embed'
+import { Tweet } from 'react-tweet'
+
+export type EmbedProps = ComponentProps<'div'> & {
+  link: string
+  linkMetadata?: LinkMetadata
+}
+
+export default function Embed({
+  link: url,
+  linkMetadata,
+  ...props
+}: EmbedProps) {
+  const Component = useMemo(() => getComponent(url), [url])
+
+  return (
+    <div {...props} className={cx('w-full', props.className)}>
+      {Component && <Component link={url} linkMetadata={linkMetadata} />}
+    </div>
+  )
+}
+
+type EmbedComponentProps = {
+  link: string
+  linkMetadata?: LinkMetadata
+}
+const urlMapper: {
+  component: React.ElementType<EmbedComponentProps>
+  checker: (link: string) => boolean
+}[] = [
+  {
+    component: ({ link }) => (
```

Comment:
Can you add special our own embed for Grill Chat?

If you share link to chat, render chat preview w/ button (like in Telegram)